src/test/java/com/intuit/graphql/orchestrator/fieldresolver/ValueTemplateTest.java

```diff
@@ -1,19 +1,34 @@
 package com.intuit.graphql.orchestrator.fieldresolver;
 
-import static org.assertj.core.api.AssertionsForInterfaceTypes.assertThat;
+import com.intuit.graphql.orchestrator.schema.transform.FieldResolverContext;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mockito;
 
 import java.util.HashMap;
 import java.util.Map;
-import org.junit.Test;
+
+import static org.assertj.core.api.AssertionsForInterfaceTypes.assertThat;
+import static org.mockito.Mockito.when;
 
 public class ValueTemplateTest {
 
-  private Map<String, Object> testDataSource = new HashMap<>();
+  private Map<String, Object> testDataSource;
+  private FieldResolverContext fieldResolverContextMock;
+
+  @Before
+  public void init() {
+    fieldResolverContextMock = Mockito.mock(FieldResolverContext.class);
+    testDataSource = new HashMap<>();
+  }
 
   @Test
   public void compile_SimpleVariable_success() {
     String valueTemplateStr = "$someVar";
-    ValueTemplate subjectUnderTest = new ValueTemplate(valueTemplateStr);
+    ValueTemplate subjectUnderTest = new ValueTemplate(fieldResolverContextMock, valueTemplateStr);
+
+    when(fieldResolverContextMock.getRequiredFields()).thenReturn(testDataSource.keySet());
```

Comment:
I would suggest create a new Set variable probably named `resolverRequiredFields`,  instead of getting from `testDataSource`.  I'm thinking the readers of this test might think the required fields is acquired from dataSource in real scenario.

Reply:
might also need to change the old test methods

Reply:
ok, I will make that change